src/app/server-cluster/ServerClusterInterfaceRegistry.h

```diff
@@ -131,6 +131,86 @@ struct LazyRegisteredServerCluster
 /// Allows registering and retrieving ServerClusterInterface instances for specific cluster paths.
 class ServerClusterInterfaceRegistry
 {
+public:
+    ~ServerClusterInterfaceRegistry();
+
+    /// Add the given entry to the registry.
+    ///
+    /// Requirements:
+    ///   - entry MUST NOT be part of any other registration
+    ///   - LIFETIME of entry must outlive the Registry (or entry must be unregistered)
+    ///
+    /// There can be only a single registration for a given `endpointId/clusterId` path.
+    [[nodiscard]] CHIP_ERROR Register(ServerClusterRegistration & entry);
+
+    /// Remove an existing registration
+    ///
+    /// Will return CHIP_ERROR_NOT_FOUND if the given registration is not found.
+    CHIP_ERROR Unregister(ServerClusterInterface *);
+
+    /// Return the interface registered for the given cluster path or nullptr if one does not exist
+    ServerClusterInterface * Get(const ConcreteClusterPath & path);
+
+    // Set up the underlying context for all clusters that are managed by this registry.
+    //
+    // The values within context will be copied and used.
```

Comment:
since this is a rvalue reference, probably not "copied" but rather moved/used as is ?